app/src/main/java/mozilla/lockbox/store/AccountStore.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.store
+
+import android.content.SharedPreferences
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subjects.PublishSubject
+import mozilla.components.service.fxa.FirefoxAccount
+import mozilla.components.service.fxa.OAuthInfo
+import mozilla.components.service.fxa.Profile
+import mozilla.lockbox.support.Optional
+import mozilla.lockbox.support.asOptional
+import mozilla.lockbox.flux.Dispatcher
+import mozilla.lockbox.support.SecurePreferences
+
+private const val FIREFOX_ACCOUNT_KEY = "firefox-account"
+private val FXA_SCOPES = arrayOf("profile", "https://identity.mozilla.com/apps/lockbox")
+
+class AccountStore(
+    private val dispatcher: Dispatcher = Dispatcher.shared,
+    private val securePreferences: SecurePreferences = SecurePreferences()
+) {
+    companion object {
+        val shared = AccountStore()
+    }
+
+    internal val compositeDisposable = CompositeDisposable()
+
+    val oauthInfo: Observable<Optional<OAuthInfo>> = PublishSubject.create()
+    val profile: Observable<Optional<Profile>> = PublishSubject.create()
+
+    init {
+        // registration for further actions happens here
+    }
+
+    fun apply(sharedPreferences: SharedPreferences) {
+        securePreferences.apply(sharedPreferences)
+
+        loadFxA()
+    }
+
+    private fun loadFxA() {
```

Comment:
Nit: there's a bunch of things I don't like about this naming. 

First: prefer `FirefoxAccount` or `Account` to `FxA`.

Second: you're not returning anything (as you might expect to `load` an existing account); you're then resaving immediately after you've loaded, or updating the profile/oauth with `null`.

Reply:
We need to resave immediately after loading the `FirefoxAccount` from the JSON string because that's when they do the account refresh.

I'm mirroring what we have in iOS here, but I'm happy to refactor if it would be clearer.